mypy/build.py

```diff
@@ -265,9 +265,11 @@ def _build(sources: List[BuildSource],
                            flush_errors=flush_errors,
                            fscache=fscache)
 
+    TypeState.reset_all_subtype_caches()
     try:
-        TypeState.reset_all()
         graph = dispatch(sources, manager)
+        if not options.cache_fine_grained or options.fine_grained_incremental:
```

Comment:
Should we actually clear the caches when using `cache_fine_grained`? I think that it only generates some additional module metadata that is not related to subtype checks.

Reply:
Indeed, we don't need it.